Anand_Dark mode for all modals on Reports page

anandsesha opened the pull request:
# Description
Implemented dark mode for all modals on the Reports page.

## Related PRS (if any):
This frontend PR is related to the dev backend


## Main changes explained:
- For functional components, accessed darkMode redux state and styled modals with conditional rendering.
- For class components accessed the darkMode state from the redux store using mapStateToProps and connect function.
- Followed [Dark Mode Implementation Guide](https://docs.google.com/document/d/11OXJfBBedK6vV-XvqWR8A9lOH0BsfnaHx01h1NZZXfI/edit#heading=h.xfui197wwd7y) for applying theme, style and managing state.

All modals that can be found on the Reports page can now be toggled to dark mode.


## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as any user
5. go to dashboard→ Reports → Reports,
6. Toggle to dark mode and verify if all modals are working in dark mode.


## Screenshots or videos of changes:
Areas where dark mode is applied to modals:

- Add Lost Time modal (For Project/Person/Team options and error color is bold in dark mode)
![Screenshot (888)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/4b07e4bf-c697-4d1d-8cd4-7e1740558660)

![Screenshot (890)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/74c4965c-3289-4623-8473-baec5babf3be)

- Show Project/Person/Team Lost Time - Edit and Delete popups
![Screenshot (886)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/ddf7eebb-914f-43bd-9efb-4d3f0d7323f6)
![Screenshot (887)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/43659736-4a4b-42c0-93b8-7969fcc6231a)

- Info tooltip modals "i" icons (8 occurrences on reports page)
![Screenshot (902)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/62a9bb88-7f8c-4bbc-9820-a64ef81eccaf)
![Screenshot (903)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/4faf52d6-1ef1-4bfc-9729-db374dc41d8e)
![Screenshot (905)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/ba5c25b8-91c8-4574-a508-6c4a416a6141)
![Screenshot (906)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/30778cf5-623c-48b6-8e4f-3447c34fbe1e)


## Note:
Theme/Layout/Color pallette can be verified here  [Dark Mode Implementation Guide](https://docs.google.com/document/d/11OXJfBBedK6vV-XvqWR8A9lOH0BsfnaHx01h1NZZXfI/edit#heading=h.xfui197wwd7y)

dkn6 reviewed (changes requested):
Thank you for your great work!
I have found a TypeError occurring. Attached is an image for your reference.

<img width="743" alt="#2250" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/168319518/39a8388b-af05-443f-a819-b5aeeb9d071f">

nathcarnevalli reviewed (commented):
I've found some bugs when changing the modes:

1- When I'm not in dark mode and I switch to dark mode, the edit and delete modal doesn't change
2 - When I'm in dark mode and I click to show, for example, 'Show Person Lost Time', if I switch modes again, the letters become difficult to see

Video:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69649453/84a852cd-6fcc-45ff-b7d2-afa5e2ee4c84

linkhanthtel reviewed (changes requested):
I reviewed your PR and all looks good! But, when I switch back to dark theme, 'Edit Lost Time' components does not change to dark theme. Kindly see the attached image.

<img width="1439" alt="#2250 " src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95618937/823bdb81-49d6-440a-b2f4-a90c6ff3b8d9">

<img width="1440" alt="#2250 (2)" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95618937/d31749d7-6e3b-4a3c-8e19-a52bd50b0706">

ZhangPeizhou commented:
![Capture](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95693786/e1deeb23-75bf-49ce-957e-9f28e07491f5)

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95693786/beac32c8-1440-4354-ac2b-11618fcac06c

For the "Edit Lost Time", after I switched back to the light mode, it remained in the dark mode, as shown in the picture. Please check.

Nahiyan-16 commented on src/components/Reports/LostTime/AddLostTime.jsx:
```diff
@@ -285,15 +288,15 @@ const AddLostTime = props => {
-    <Modal isOpen={props.isOpen} toggle={props.toggle}>
-      <ModalHeader toggle={props.toggle}>
+    <Modal isOpen={props.isOpen} toggle={props.toggle} className={darkMode ? 'text-light dark-mode' : ''}>
+      <ModalHeader toggle={props.toggle} className={darkMode ? 'bg-space-cadet text-light' : ''}>
```
Since we already have the classname, 'text-light', within the parent component, it is unnecessary for us to declare it again on ModalHeader, ModalBody, or ModalFooter

Nahiyan-16 commented on src/components/Reports/LostTime/AddLostTime.jsx:
```diff
@@ -285,15 +288,15 @@ const AddLostTime = props => {
           <FormGroup>
-            <Label>Type</Label><br/>
-            <div className='type-container'>
+            <Label for="entryType" className={darkMode ? 'text-light' : ''} >Type</Label><br/>
```
Since we are going to be repeating this line of code throughout this file, it is better to declare a variable like `const fontColor = darkMode ? 'text-light' : ' ';` and replace it within our classnames like so `className={fontColor}`

Nahiyan-16 commented on src/components/Reports/LostTime/EditHistoryModal.jsx:
```diff
@@ -88,7 +89,7 @@ const EditHistoryModal = props => {
       return (
         <FormGroup>
-          <Label>Project Name</Label>
+          <Label className={darkMode ? 'text-light' : ''}>Project Name</Label>
```
Same recommendation as above. If we are repeating lines of code 2 or more times it's better to have it stored in a variable.

Nahiyan-16 commented on src/components/Reports/LostTime/EditHistoryModal.jsx:
```diff
@@ -403,27 +404,27 @@ const EditHistoryModal = props => {
-      <ModalFooter>
-        <Button onClick={handleCancel} color="danger" style={boxStyle}>
+      <ModalFooter className={darkMode ? 'bg-yinmn-blue' : ''}>
+        <Button onClick={handleCancel} color="danger" style={darkMode ? boxStyleDark : boxStyle}>
```
Same recommendation as above. In order to make our code more DRY, we can declare a variable like `const boxStyling = darkMode ? boxStyleDark : boxStyle;` and use it like so `style={boxStyling}

Nahiyan-16 reviewed (changes requested):
Hey Anand, you did some awesome work here, especially with changing the modal header close button to toggle between dark and light mode. I also like how you added color to the Save and Close buttons. 

I do however request some slight changes so we can keep our codebase tidy and DRY.

Gayathr-2012 commented:
Hi Anand,
I checked your PR and the dark mode is working fine. I also checked the "Add Lost Time" feature, and it works well with dark mode. However, the date field is not displaying correctly in dark mode. Could you please fix this?

![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/bf77688a-781b-40a9-a183-b55debcdd51e)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/b944f174-cdaf-4580-9e87-cabc1865d554)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/749e2871-76e9-4e60-b7ef-766236b68ea7)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/0dc7de1d-422f-42ac-84cc-3f6dfee326a6)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/0695aba3-86a2-4b64-ae90-99e80aa3e362)


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/5e29727e-955b-4f92-a6ed-d62efdfb9bfa

Gayathr-2012 reviewed (changes requested):
Hi Anand,
I checked your PR and the dark mode is working fine. I also checked the "Add Lost Time" feature, and it works well with dark mode. However, the date field is not displaying correctly in dark model. In dark mode, the date picker is in light mode.. Could you please fix this?

![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/bf77688a-781b-40a9-a183-b55debcdd51e)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/b944f174-cdaf-4580-9e87-cabc1865d554)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/749e2871-76e9-4e60-b7ef-766236b68ea7)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/0dc7de1d-422f-42ac-84cc-3f6dfee326a6)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/0695aba3-86a2-4b64-ae90-99e80aa3e362)


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/163784958/5e29727e-955b-4f92-a6ed-d62efdfb9bfa

hbodgal reviewed (dismissed):
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/145157488/d8d3b389-d5de-4c4c-ba2b-45abd77d1d85)

can you center the "Close" button. and remove the scroll bar.

Sandhya1236 reviewed (dismissed):
Everything appears to be functioning correctly, as I have thoroughly tested it and requested changes have been implemented. However, I noticed that the date feature is not visible in dark mode while the rest of the page is. If it's intended to be visible, it should be updated accordingly.
![Screenshot (691)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/79351059/e2e5e6aa-e509-42af-9831-025c6b41f3ef)

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/79351059/99c3dcee-5dbd-4c86-b7b7-df4bc2f1e808

nishithaShetty commented:
Hi Anand,
Great work! Everything works as expected. I tested it as an owner and admin

- The screenshots are from the owner account.
- The scope shows the modal, but as mentioned earlier, the date picker is still light.
<img width="727" alt="PR#2250" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/45133470/26eb38f4-20c8-42c4-8994-d7e3826b502e">
<img width="588" alt="PR#2250_1" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/45133470/0a456f9b-18be-42c6-b71b-2c33a88ef17e">
<img width="483" alt="PR#2250_2" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/45133470/c29382eb-a5d6-4498-a757-4da74093e4eb">
<img width="520" alt="PR#2250_3" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/45133470/d98cda06-aa08-4d70-b77a-327e7f39cef3">
<img width="909" alt="PR#2250_4" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/45133470/b2b75a55-e254-40be-9ad4-1616f5044d63">
<img width="648" alt="PR#2250_5" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/45133470/02a5bc82-9d22-44a7-8db7-3bcfc90d8820">

KaushikMreddy commented:
Hi Anand, all your changes look good and are working fine. Good work!

jennZhang93 commented:
Did `npm run start:local` but encountered the following error. Seems like the script is missing a closing tag.
<img width="721" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/66922622/04c69757-ab1c-4bf3-a050-332ab6005a6f">

KyreneAF reviewed (changes requested):
- After running the frontend an error occurs on line 301. This can be fixed by adding another ModalFooter closing tag as seen on line 300.

<img width="491" alt="2250 error" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/129882345/9cb51de5-a0f3-4dc7-b987-caf74e58a13b">

- The information modals work as expected. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/129882345/86f8ff28-22d6-4356-ad79-eb6107f5290d

- I am unable to view the Add lost time button that opens the modal.

![unable to see add lost time button](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/129882345/e4d7f23c-4867-4203-8704-cde85dbca718)

JatinAgrawal94 reviewed (changes requested):
A Modal Footer closing tag needs to be added. Rest else works fine. Great Work!
<img width="1222" alt="PR#2250" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47699187/22a40acf-bea6-47e3-9dbc-15d52d493fd4">

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47699187/1f5d8bc5-93bc-413f-a8ac-9f4b68197449

anandsesha commented:
Changes:
- Date field associated with Start Date and End Date handled in Dark Mode.
- Info modal close button and heading alignment fixed.
- Modular and DRY code: removed redundant textColor usage, centralized fontColor and boxStyling logic with utility functions and updated respective components.

Both Issues: EditLostTime modal not switching to dark mode and, Modal Footer missing closing tag are not observed. Kindly retest.

![Screenshot (922)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/166a4c18-5c16-4a28-aca7-11121248b7e3)
![Screenshot (924)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78548214/2e12a2ab-3e35-4f5c-a7d8-f2c24ccc0ece)

jinxyou reviewed (dismissed):
I tested this PR on my setup and all the modals looks great under the dark mode. Great work!
![Screenshot 2024-06-01 172716](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13797542/bdd6a827-e0b4-4fa8-9678-6023b1ea1ec8)

CloudyZ524 reviewed (commented):
Hi Anand,
I've switched to your branch and tested the dark mode features you developed in this pull request. This pages rendered successfully in dark mode without error. However, I cannot see the "Add Lost Time" button mentioned in your instructions as well as other reviews. Please take a look and make sure the latest commit includes all new updates. 

<img width="1336" alt="Screenshot 2024-06-06 at 13 05 41" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/124476559/12948804-66bb-493c-a4d4-6003152999ac">

Wenboooooo reviewed (approved):
Hi, the dark mode on the reports page works as expected. All modals are functioning. Great work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114895283/26a003a9-512b-4ee2-ab84-f661f81940f0

akshit0211 commented:
Hi Anand, the dark mode works as expected. But, I can't see the add lost time button on the reports page. 

<img src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112666943/9006eb1a-954f-4e8a-8c1f-33ab5adeda07" alt="s1">
<img src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112666943/54f35ada-f78e-4dee-84ae-fe88acde1b89" alt="s2">

Kavil-Jain-514 reviewed (dismissed):
The dark mode is working well for all the modals.
However, when I clicked the show details button, the text color was still black in dark mode and didn't look good. Can you change that to white and replace the hover color with something lighter?

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112303944/43118d89-4522-4868-aba6-eac219fae90b

Sudershhh reviewed (dismissed):
Hey Anand,

The Dark mode is working as expected for the modals.

![PR 2250](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/59435391/5889be30-a3e7-4971-90e8-ee57f9555065)

suaniii reviewed (approved):
I checked your PR, and the dark mode is working fine. I verified the "Add Lost Time," "Show Project/Person/Team Lost Time" features, and the "Info tooltip modals" icons, and they all function well with dark mode. Additionally, the date fields associated with `Start Date` and` End Date` display correctly in dark mode, and the info modal close button and heading are centered. Great work!


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113208952/d4114e64-5393-44e2-a7c1-b64a260ea1a3

Jingyii800 reviewed (approved):
Well done. The dark mode has been implemented to all modals in reports page.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112589476/ad14896f-6daa-428f-bacd-79b440d58460

ImzIssa reviewed (dismissed):
Hey, the dark mode applies on the modals. great work.
![Screenshot (407)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/ba409cea-4a74-463d-97ef-5a9a61d422d1)
![Screenshot (408)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/88bff2cf-2177-49de-8c02-f4b4a595571d)

However, This table needs some work to reflect the dark mode 
![Screenshot (406)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/28c586dd-ec29-4ed0-83bb-26b84ed26705)

AnujVakil reviewed (changes requested):
I have tested this PR in dark mode and it looks fine. Just need to align close button in center for every info section.

<img width="1432" alt="PR #2250_1" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118478795/ee00384d-b9be-4e17-a302-746eb17251f8">



<img width="1438" alt="PR #2250_2" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118478795/38ce331f-6f70-4c59-99b6-ca152f3e488f">

<img width="1440" alt="PR #2250_3" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118478795/3a5d6146-c81d-4d8e-a088-783b335d986c">

<img width="1437" alt="PR #2250_4" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118478795/fd6f8f1e-bd77-4b63-9420-0a43b857f525">

<img width="1436" alt="PR #2250_5" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118478795/e2a0fc97-c2b6-4b36-b4d6-a6abb4686ed0">

Vigneshwar-Muriki reviewed (commented):
I checked out the current branch in the frontend and the dev branch in the backend, ran the commands, and logged in as an admin user to test this PR. When I navigated to the reports page, I was unable to see the "Add Lost Time" button. The video below has the details.

https://github.com/user-attachments/assets/61045b84-a5dc-402d-8b95-2b44af19e846

Shadhrush5 reviewed (dismissed):
Hi, I reviewed PR #2250 with the current branch in the frontend and the dev branch in the backend.
I ran the commands to install and build
Logged in as an admin user
Navigated to Dashboards -> Reports -> Reports
In reports page, I was unable to see the "Add Lost Time" button. I have attached a picture for the same.

<img width="1728" alt="Screenshot 2024-07-17 at 12 41 52 PM" src="https://github.com/user-attachments/assets/773686d3-d3c2-44ad-9535-01eb2837d6f2">

SrikanthChowdary495 reviewed (commented):
I have tested the following points. Everything is working so perfectly in both dark mode & light mode. 
1. Add Lost Time modal (For Project/Person/Team options and error color is bold in dark mode)
2. Show Project/Person/Team Lost Time - Edit and Delete popups
3. Info tooltip modals "i" icons (8 occurrences on reports page)

https://github.com/user-attachments/assets/328b430d-e315-4383-9b58-ac88472aca00

SaiLavanya1 reviewed (commented):
Hi,
I have reviewed the PR #2250 for implementing dark mode for all modals on the Reports page. However, I encountered an issue where the "Add Lost Time" button is not visible on the Reports page. I have followed the provided steps to check out the branch, install dependencies, clear site data/cache, and run the application, but the button does not appear. Could you please check if there might be an issue causing the button not to display?
<img width="1445" alt="Screenshot 2024-07-24 at 2 18 13 PM" src="https://github.com/user-attachments/assets/53c908be-3f8e-4ef8-9e0c-b4646d5c6ce3">
<img width="1457" alt="Screenshot 2024-07-24 at 2 18 24 PM" src="https://github.com/user-attachments/assets/d5d2f127-604f-46b5-9626-b2cf1a24443c">

Pranay-ai reviewed (approved):
Hey @anandsesha 
I have Revied the PR and Switched between dark Mode and Light mode it works fine.
![image](https://github.com/user-attachments/assets/51c7ddc4-8288-49a5-a7a3-3d5437bbbb30)

Thank You.

ReinaT5678 reviewed (changes requested):
I reviewed the dark mode for all modals on the report page, and everything seems to be working as expected! I did, however, run into an issue where the text disappears under the project lost time section in light mode after switching from dark mode. I attached a video of it below:

https://github.com/user-attachments/assets/93f20eff-6295-4cef-b074-89cfba71bdaf

arunsrao9 commented:
I tested the dark mode for all modals on the report page, and everything seems to be working as expected. I didn't face any issue with the project lost time section as well. Great work!
<img width="1440" alt="Screenshot 2024-08-02 at 10 52 53 PM" src="https://github.com/user-attachments/assets/fcdbd566-a5af-4141-b609-74e45ab52b4f">
<img width="1440" alt="Screenshot 2024-08-02 at 10 52 44 PM" src="https://github.com/user-attachments/assets/04b108b5-942b-4835-8791-454a096368db">
<img width="1440" alt="Screenshot 2024-08-02 at 10 52 37 PM" src="https://github.com/user-attachments/assets/4db4fa33-94fa-48cb-b1b8-82f8d0b97911">

Lemonnycodes commented:
Hi,
I tested the changes introduced in this PR. The dark mode implementation for modals across the Reports page works as expected. 

Edit and Delete:
![PR 2250 (1)](https://github.com/user-attachments/assets/732a33ac-a22f-489e-8464-dac094e91e65)
![PR 2250 (2)](https://github.com/user-attachments/assets/68960d0d-e99a-4817-b168-973066340579)

"i" icon : 
![PR 2250 (3)](https://github.com/user-attachments/assets/4ed1a934-2f3f-47c3-83fd-08fa5463e9b6)

AnkitLall reviewed (approved):
I have tested the feature, following the instructions in the PR, and it works as expected. Please refer the screenshots attached below.

<img width="1512" alt="Screen Shot 2024-08-17 at 3 31 06 PM" src="https://github.com/user-attachments/assets/a546bb29-3af3-426d-97a4-a184244ab46b">
<img width="1512" alt="Screen Shot 2024-08-17 at 3 31 13 PM" src="https://github.com/user-attachments/assets/2a315f89-147a-4e72-9243-2470868be3f3">
<img width="1512" alt="Screen Shot 2024-08-17 at 3 31 26 PM" src="https://github.com/user-attachments/assets/ef4dc1a2-62d2-4112-bf11-91cdca4b1265">

adityasure9 reviewed (approved):
The PR successfully implements dark mode for all modals on the Reports page. The modals function as expected in both light and dark modes. 

![image](https://github.com/user-attachments/assets/f5503cd7-b154-48e1-86a0-6b4c2ce370bc)

Lemonnycodes reviewed (dismissed):
Hi,
I tested the changes introduced in this PR. The dark mode implementation for modals across the Reports page works as expected. 

Edit and Delete:
![PR 2250 (1)](https://github.com/user-attachments/assets/732a33ac-a22f-489e-8464-dac094e91e65)
![PR 2250 (2)](https://github.com/user-attachments/assets/68960d0d-e99a-4817-b168-973066340579)

"i" icon : 
![PR 2250 (3)](https://github.com/user-attachments/assets/4ed1a934-2f3f-47c3-83fd-08fa5463e9b6)

anandsesha commented:
Requested issues fixed:
- Project/Person/Team lost time option's text is visible when switched from Dark to light mode and back.
- Add Lost TIme button visible when modes are switched.
- Info Modal close btn, and text center aligned
- missing modal footer added

AnujVakil reviewed (dismissed):
Working as expected!

JatinAgrawal94 reviewed (dismissed):
Changes work as expected. All errors are fixed. Good work.
<img width="1502" alt="Screenshot 2024-09-02 at 12 41 55 AM" src="https://github.com/user-attachments/assets/c317765d-eaf9-4665-bbbc-68113804d0cd">

ReinaT5678 reviewed (dismissed):
Everything works as expected! I can view the text in dark mode now!

strallia reviewed (changes requested):
Some issues I found using Admin account:
- For the Total Project Report section, the Show Details table has text color that is hard to read: normal text is black on dark blue background, and when hovering over a row, the white hyperlink text cannot be read on the white background. This issue is also seen in the Total People Report and Total Team Report views. I suggest more color contrast for readability.

https://github.com/user-attachments/assets/6a59d558-bc6e-4096-9f32-7fe6949149f8

- I see the Date Picker modal on the main page is in dark mode but when I go to Show Project Lost Time > Edit one of the rows, the Date Picker there is in light mode. Issue also seen in Person Lost Time and Team Lost Time.

https://github.com/user-attachments/assets/22f30f09-1cda-4451-a734-90c5902f52d2

swaroop-2007 reviewed (changes requested):
@anandsesha 
I reviewed your PR and checked for the features in the dark mode using an admin account. 
The font color for Total Logged Time in Show Total Project Report and Total People Report is not visible until I hover on it. I have attached screenshots below.
![Screenshot (368)](https://github.com/user-attachments/assets/8d378df4-2b56-4e05-b6eb-cc18a960886a)
![Screenshot (367)](https://github.com/user-attachments/assets/4684b46e-2811-4ca2-9880-42cad57a6f5d)
![Screenshot (366)](https://github.com/user-attachments/assets/121db76a-19ba-450d-affe-24d65e1dc1c5)

yashwanth170 reviewed (dismissed):
Everything is working as expected! I see most of the issues mentioned above are already fixed since I didn't notice any
![dar_mode_1](https://github.com/user-attachments/assets/7ad127be-04cc-4215-b3e1-662e162c845f)

anandsesha commented:
The font color and hover issue for the Team/People/Project report details table in dark mode have been fixed. 

@strallia 
Please note the following regarding your second issue with the DatePicker: 

The Date Picker in the 'Add Lost Time' form cannot be styled exactly like the one in the Reports page because the input field in 'Add Lost Time' uses the default browser calendar, which doesn’t allow custom styling the way we can with react-datepicker. This limitation comes from the inherent difference between the native input element of type date and a fully customizable DatePicker component.

To ensure consistency across different pages that display a calendar, we could either convert all such input fields to use a library like react-datepicker or another similar library that allows custom styling. 

I suggest we track this effort as a separate issue, where we can decide on a consistent implementation for dark mode across all date inputs.

swaroop-2007 reviewed (dismissed):
@anandsesha 
Nice work! The requested changes are functioning as expected, so I’m approving the PR.

strallia reviewed (changes requested):
> @strallia Please note the following regarding your second issue with the DatePicker:
> 
> The Date Picker in the 'Add Lost Time' form cannot be styled exactly like the one in the Reports page because the input field in 'Add Lost Time' uses the default browser calendar, which doesn’t allow custom styling the way we can with react-datepicker. This limitation comes from the inherent difference between the native input element of type date and a fully customizable DatePicker component.
> 
> To ensure consistency across different pages that display a calendar, we could either convert all such input fields to use a library like react-datepicker or another similar library that allows custom styling.
> 
> I suggest we track this effort as a separate issue, where we can decide on a consistent implementation for dark mode across all date inputs.
> 
> The font color and hover issue for the Team/People/Project report details table in dark mode have been fixed. Screenshots below.
> 
> ![Screenshot (1048)](https://private-user-images.githubusercontent.com/78548214/365846337-49183d9b-8150-4fe4-b567-da5a740f17aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5OTU0NjksIm5iZiI6MTcyNTk5NTE2OSwicGF0aCI6Ii83ODU0ODIxNC8zNjU4NDYzMzctNDkxODNkOWItODE1MC00ZmU0LWI1NjctZGE1YTc0MGYxN2FhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDE5MDYwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyMjkxODQ3YTI1ZTk0MTYyYjRjZmVlZGU3MDhiZDMzYTkxOGRjOGUxODhjNThiZmZlNjEyYjUwZGJiNTlmYTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GWjetFID8l14g3jZrktwUuySMR-fkYVfa3EVeBw2hiw)

About the date picker styles--that makes sense. I agree that style consistency there should be a separate PR/issue.

As for the font color changes, they are working on the Total Project/People Report pages. But it looks like it messed with color contrast on the Team page.   

Before applying commit `11bc61e Team/People/Project report details table dark theme hover`:  

https://github.com/user-attachments/assets/90377da2-5d02-452b-a76e-2bc9b87ad34b

After applying the commit: 

https://github.com/user-attachments/assets/c2ee458f-8a24-4642-9c4f-fa902c327934

strallia reviewed (dismissed):
@anandsesha The changes work. Nice job!

<img width="1167" alt="Screenshot 2024-09-13 at 3 44 26 PM" src="https://github.com/user-attachments/assets/5553b6d6-b6da-40c9-ab06-2dcb1ecfe353">

crystallow1168 reviewed (dismissed):
Hi @anandsesha,
Changes looks good! Below are my findings:

https://github.com/user-attachments/assets/b07ebf84-eee7-47bd-92e6-b6f5e6585144

✅ Add Lost Time modal (For Project/Person/Team options and error color is bold in dark mode)


<img width="888" alt="image" src="https://github.com/user-attachments/assets/0a30acf9-6596-4b39-9c5c-964e1af0e10e">

✅ Show Project/Person/Team Lost Time - Edit and Delete popups in dark mode


https://github.com/user-attachments/assets/f4aff5cc-6aee-4feb-a736-085effd50984

✅ Info tooltip modals "i" icons (8 occurrences on reports page) in dark mode


<img width="686" alt="image" src="https://github.com/user-attachments/assets/3e17003e-74b0-45f3-9d46-c33b45975cdb">

✅ Date field associated with Start Date and End Date handled in Dark Mode.

<img width="1322" alt="image" src="https://github.com/user-attachments/assets/6fed6e64-9ba2-480c-90dc-3d334a5fd688">

✅ Total Project Report, Total People Report, and Total Team report graph looks good as well

SFA23SCM35V reviewed (approved):
All modals on the Reports page now support toggling to dark mode.
![1](https://github.com/user-attachments/assets/db4dd027-a8ec-4e1c-ba16-55089be7bc87)
![2](https://github.com/user-attachments/assets/ba3e3671-8ea5-4991-a383-214774810ec0)
![3](https://github.com/user-attachments/assets/a73cdbce-cdb3-4df4-b082-3c41eaca9629)
![4](https://github.com/user-attachments/assets/b810b8b3-fcef-4180-8e7c-e074c287ec98)
![5](https://github.com/user-attachments/assets/8b00ce5a-d961-4f4f-ae33-3996b7748ed0)
![6](https://github.com/user-attachments/assets/1628c9eb-6c56-435d-8862-8c78a4c90cdf)
![7](https://github.com/user-attachments/assets/b1a42ab1-861c-4acf-b9bc-696b3fe83b1a)

neeharikakoniki reviewed (approved):
The transition between light and dark modes on the Reports page modals is smooth and seamless. The conditional rendering is well done, ensuring that all modals adapt to the dark mode setting without any visual or functional glitches

Before 
![image](https://github.com/user-attachments/assets/d676daa0-1d5b-4529-8e85-6989c02744c6)

After 
![image](https://github.com/user-attachments/assets/f83367be-40d9-4aa2-ae4b-2f0b97d2f97f)

![image](https://github.com/user-attachments/assets/45a30bcd-2bb9-425c-bf2e-6b3780e722ce)
![image](https://github.com/user-attachments/assets/b179d4f1-11c1-4af6-9a4f-618bf9920794)

anirudhsk2107 reviewed (dismissed):
Hi @anandsesha

The changes are working as expected. 
- All pop ups on 'i' button are in dark mode
- Start Date and End Date  
![Screen Shot 2024-09-20 at 6 35 56 PM](https://github.com/user-attachments/assets/0866d592-cb88-4034-ab2c-67127aaf1528)
![Screen Shot 2024-09-20 at 6 35 52 PM](https://github.com/user-attachments/assets/a9402259-5b05-4b0e-ba94-ac35880a5c44)
- Total Team Report table
![Screen Shot 2024-09-20 at 6 37 31 PM](https://github.com/user-attachments/assets/cdc894e4-cfed-4c11-8f83-21174f7acf5a)
- Edit and Delete Lost time
![Screen Shot 2024-09-20 at 6 00 54 PM](https://github.com/user-attachments/assets/2e4287c9-cb27-4bc6-b6e0-4ea496781967)
![Screen Shot 2024-09-20 at 6 00 37 PM](https://github.com/user-attachments/assets/37265f96-b898-4fa9-8069-de3a2a973110)

saurabhshetty29 commented:
The transition between light and dark modes on the Reports page modals is fluid and seamless. The conditional rendering is executed effectively, ensuring that all modals adjust to the dark mode setting without any visual or functional issues.
<img width="1414" alt="Screenshot 2024-09-24 at 10 35 23 PM" src="https://github.com/user-attachments/assets/003aab26-cfe8-4fe0-b64f-e241f6932b1a">

mrinalini-raghavendran19 reviewed (approved):
Looks good to me!

Here are some examples as to where the modals are working great in dark mode:

<img width="1470" alt="Screenshot 2024-09-26 at 3 43 49 PM" src="https://github.com/user-attachments/assets/60f96571-8a5a-48e4-b6c2-8ea3eec44d58">

<img width="1470" alt="Screenshot 2024-09-26 at 3 43 53 PM" src="https://github.com/user-attachments/assets/f2f90c48-ce48-4ca1-832f-39d17c942551">

However, a suggestion - when I try to edit lost time in the 'Edit Lost Time' Modal, the calendar appears in light mode; which can be fixed as in the filter selection div:

<img width="1470" alt="Screenshot 2024-09-26 at 3 39 07 PM" src="https://github.com/user-attachments/assets/a4a8f044-928f-4ab3-b3d6-acef88ef6bff">

I have logged in as an 'admin' but I don't see the Add lost time button - do I need to login through another type of account?  (the reports section doesn't appear for volunteer or manager accounts)

lcohen730 reviewed (dismissed):
Dark mode works successfully for all modals tested: Show Project Lost Time - Edit and Delete popups, Info tooltip modals "i" icons, and date pickers.


https://github.com/user-attachments/assets/21877d14-8d84-49fb-a56a-23ef4563d140

Lemonnycodes reviewed (approved):
Hi Anand,

I have reviewed the PR and it works perfectly. 

- Tested  Reports page, including the Add Lost Time modal and the Show Project/Person/Team Lost Time popups, and they all toggle to dark mode as expected.
-  The bold error color in dark mode adds clarity, and the info tooltip modals function correctly in the dark mode.
- Date field associated with Start Date and End Date is handled well in Dark Mode.

Screenshots:
![PR 2250 (1)](https://github.com/user-attachments/assets/186366fd-2dbd-4d48-8c60-905785705ca0)
![PR 2250 (2)](https://github.com/user-attachments/assets/ded39494-53c0-4d36-8078-94e97c34f2f3)
![PR 2250 (3)](https://github.com/user-attachments/assets/29dd9032-7b47-43ab-9968-6e05dd38a273)
![PR 2250 (4)](https://github.com/user-attachments/assets/53bb1d85-c0b6-4507-b24d-646cffa26f35)
![PR 2250 (5)](https://github.com/user-attachments/assets/d8952be8-1685-4ebe-a0f8-28814eea1a6d)
![PR 2250 (6)](https://github.com/user-attachments/assets/ba7d3b01-2f90-4639-b366-dd4383051c2c)
![PR 2250 (7)](https://github.com/user-attachments/assets/5abe8e21-4128-4bc8-8957-eaf0136ae901)
![PR 2250 (8)](https://github.com/user-attachments/assets/51d09952-3718-4ccb-85a2-cfbf9be2ee63)
![PR 2250 (9)](https://github.com/user-attachments/assets/6a84264e-d031-4fc2-b303-7e9753657828)

one-community commented:
Thank you all, merging!
